meta/Meta.cpp

```diff
@@ -7446,18 +7446,31 @@ void Meta::meta_sai_on_fdb_event_single(
                 sai_attribute_t *list = data.attr;
                 uint32_t count = data.attr_count;
 
-                sai_attribute_t local[2]; // 2 for port id and type
-
-                if (count == 1)
+                //Add an attr of type if possible.
+                sai_attribute_t local[SAI_FDB_ENTRY_ATTR_END]; // 2 for port id and type
+                if (count <= SAI_FDB_ENTRY_ATTR_END)
                 {
-                    // workaround for missing "TYPE" attribute on notification
-
-                    local[0] = data.attr[0]; // copy 1st attr
-                    local[1].id = SAI_FDB_ENTRY_ATTR_TYPE;
-                    local[1].value.s32 = SAI_FDB_ENTRY_TYPE_DYNAMIC; // assume learned entries are always dynamic
-
-                    list = local;
-                    count = 2; // now we added type
+                    bool hasType = false;
+                    for(uint32_t i=0; i<count; i++)
```

Comment:
please add spaces to make code more readable, please take a look how other loops are crafted